db/migrate/20240919034213_drop_microposts.rb

```diff
@@ -6,9 +6,10 @@ def up
   def down
     create_table :microposts do |t|
       t.text :content
-      t.references :user, foreign_key: true
+      t.references :user, null: false, foreign_key: true
 
       t.timestamps
     end
+    add_index :microposts, [:user_id, :created_at]
```

Comment:
Sau khi code đã merged vào master rồi thì ko edit migration file e nhé, cả cái null ở trên cũng thế, muốn change cái gì trong DB phải tạo file migrate mới
File migrate này đã đc chạy rồi, e edit vào thì dùng lệnh migrate nó đâu chạy lại nữa

Reply:
Dạ do khi trước em nhớ a kêu gom mấy migration của micropost lại 1 cái e ko tạo mới sợ nhiều cái, xong e vào migrate status để bật down mấy migrate cần chỉnh lại, xong chỉnh rồi chạy migrate để nó áp dụng các migrate đó up lên lại.

Reply:
Lúc trc a kêu gom là lúc đó tạo mới migration, chưa merge master, ở local thì e có thể edit, chỉnh sửa thoải mái
Nhưng hiện tại code file migration micropost cũ đã đc merge vào master, trong thực tế thì nó sẽ được deploy lên production. Trên đó thì e ko thể tự do xoá, sửa như ở local mà tất cả phải thực hiện qua migration

Reply:
Dạ em có khôi phục lại rồi a, mà có cái dấu cách bữa setting auto format lại như vậy được không ạ, không ảnh hưởng đến cấu trúc migrate

Reply:
> có cái dấu cách

Ko sao e nhé, ko thêm code là đc, còn dấu cách, dòng trống thoải mái

Reply:
Ko phải drop table rồi tạo lại đâu e nhé, drop thì trên production nó mất hết data thì sao
E tham khảo guide này để add index:
https://www.beekeeperstudio.io/blog/rails-migration-add-index
Add contraints null:
https://stackoverflow.com/questions/62944480/rails-migration-add-a-null-false-column-with-an-initial-value-thats-not-an-ong

Có thể gộp 2 cái trên vào 1 file migration mới

Reply:
Dạ e có chỉnh lại thêm index và contraints chớ ko xóa bảng lại để thử cho đúng